### PR TITLE
Autostart

### PR DESCRIPTION
**Pull request by drewmoseley**

### CODE REVIEW

**Comment by drewmoseley** [comment]
This is not yet ready for merge but it is ready to be reviewed.

File: image/meta-gcp-iot/recipes-gcp/python-docs-samples/files/start-mqtt-example.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export REGION_ID='us-central1'
+
+export REGISTRY_ID='mender-demo'
+export DEVICE_ID='mender_ota_rasp3'
+
+python cloudiot_mqtt_example.py \
+       --project_id "${PROJECT_ID}" \
+       --registry_id "${REGISTRY_ID}" \
+       --device_id "${DEVICE_ID}" \
+       --algorithm RS256 \
```
Comment (drewmoseley) [inline image/meta-gcp-iot/recipes-gcp/python-docs-samples/files/start-mqtt-example.sh:14]: @Kcr19 should this parameter be RSA256 or is RS256 correct?

**Comment by drewmoseley** [comment]
Note that the start up script still needs to be parameterized for GCP cloud project and such.

File: image/meta-gcp-iot/recipes-gcp/python-docs-samples/files/start-mqtt-example.sh (hunk 1 above)
Comment (Kcr19) [reply image/meta-gcp-iot/recipes-gcp/python-docs-samples/files/start-mqtt-example.sh:14]: RS256 should work

Comment (ptone) [reply image/meta-gcp-iot/recipes-gcp/python-docs-samples/files/start-mqtt-example.sh:14]: with the activation agent - I wasn't initially sure if we still wanted the sample. But I guess it is nice to have a step that the user sees data can be published.

We only allow one MQTT connection per device - so we should probably modify the activation agent script to also check for /data/mender/pre-auth true and quit

Comment (drewmoseley) [reply image/meta-gcp-iot/recipes-gcp/python-docs-samples/files/start-mqtt-example.sh:14]: @ptone I can remove the sample script if there is no value in it; I wasn't sure if your activation agent was intended to replace it.  Let me know your preference and I'll be sure to update this PR before merging.

**Comment by drewmoseley** [comment]
@ptone @Kcr19 I've updated this a bit.  Using _only_ the state script to hold of server connections did not work since the Mender certificate only gets generated after that script returns. I added a systemd pre-exec script that manually creates it and is always invoked before the Mender client is actually launched.  That seems to work and now the certificate is created and the mender client sits waiting for the creation of the preauth_true file.

We still need to make the project ID and such configurable.

**Comment by drewmoseley** [comment]
@Kcr19 @ptone I have added some logic to make the PROJECT_ID and other GCP variables configurable.  At build time they are stashed in /opt/gcp/etc/gcp-config.sh.  The activation agent and sample python start script pull the data from there.

As far as I can tell, this PR is ready for review and testing through a full build/test cycle.

**Comment by drewmoseley** [comment]
I think this one is ready.  Any comments?

**Comment by Kcr19** [comment]
Looks good to me. Do I do PR against this branch "Autostart" for testing
out ?

**Comment by drewmoseley** [comment]
You should not need another PR for testing. If you manually download the build script and modify it to checkout from my fork and branch instead of yours, that should suffice for testing. The alternative is to simply merge this and then fix any issues in subsequent commits.

**Pull request merged**